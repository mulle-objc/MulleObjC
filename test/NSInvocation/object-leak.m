#import <MulleObjC/MulleObjC.h>


@interface Foo : NSObject
{
   struct mulle_array   _invocations;
}
@end


@implementation Foo

- (instancetype) init
{
   struct mulle_allocator   *allocator;

   allocator = MulleObjCInstanceGetAllocator( self);
   mulle_array_init( &_invocations,
                     0,
                     &MulleObjCContainerRetainKeyCallback,
                     allocator);
   return( self);
}

- (void) finalize
{
   mulle_array_reset( &_invocations);
   [super finalize];
}

- (void) dealloc
{
   mulle_array_done( &_invocations);
   [super dealloc];
}

- (void) printUTF8String:(char *) s
{
   printf( "%s\n", s);
}


- (void) addInvocation:(NSInvocation *) invocation
{
   [invocation retainArguments];
   mulle_array_add( &_invocations, invocation);
}

@end


int   main( void)
{
   Foo            *foo;
   NSInvocation   *invocation;

   @autoreleasepool
   {
      foo        = [Foo object];
      invocation = [NSInvocation mulleInvocationWithTarget:foo
                                                  selector:@selector( printUTF8String:), "Hello"];
      [foo addInvocation:invocation];
      [foo mullePerformFinalize];
   }
   printf( "We're going in, we're going down\n");
   return( 0);
}


