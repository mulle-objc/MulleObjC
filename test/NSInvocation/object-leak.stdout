We're going in, we're going down
